### PR TITLE
Advance scheduled change to PendingApproval

### PR DESCRIPTION
**Pull request by pauljamescleary**
When a scheduled change is submitted, if there are no hard errors advance to PendingApproval status.

* `BatchChange` - changed the calculation of the batch change status; if it is pending approval and the scheduled time is set it will be `Scheduled`
* `MySqlBatchChangeRepository` - updated the de-serialize to consider scheduled time so when "getting" a batch change, the status returned will appropriately be `Scheduled`
* `BatchChangeService` - updated the `buildResponse` method to consider scheduled time.  If no fatal errors, manual review enabled, and scheduled then goto PendingApproval

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 1]
```diff
@@ -362,8 +366,13 @@ class BatchChangeService(
       .flatMap(_.toList)
 
     val allNonFatal = allErrors.forall(!_.isFatal)
+    val isScheduled = batchChangeInput.scheduledTime.isDefined && this.scheduledChangesEnabled
 
-    if (allErrors.isEmpty) {
+    // If someone says "don't manual review" but "schedule it", right now we ignore the former and
+    // just goto manual review.
+    val gotoManualReview = manualReviewEnabled && allNonFatal && (allowManualReview || isScheduled)
+
```
Comment (pauljamescleary) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: this is an interesting logic decision:

Someone posts a batch change with `allowManualReview=false`, but it has a scheduled time set on it.  What do we do?

1. Chosen solution - here we ignore the `allowManualReview`

2. Alternative - we could add an if/else here that says `if allowManualReview == false && isScheduled then throw error`

Wonder what people's thoughts are.

File: modules/core/src/main/scala/vinyldns/core/domain/batch/BatchChange.scala [hunk 2]
```diff
@@ -54,21 +60,23 @@ case class BatchChange(
...
+      case BatchChangeApprovalStatus.PendingApproval if isScheduled => BatchChangeStatus.Scheduled
       case BatchChangeApprovalStatus.PendingApproval => BatchChangeStatus.Pending
       case BatchChangeApprovalStatus.ManuallyRejected => BatchChangeStatus.Failed
       case _ =>
         (hasPending, hasFailed, hasComplete) match {
           case (true, _, _) => BatchChangeStatus.Pending
-          case (_, true, true) => BatchChangeStatus.PartialFailure
-          case (_, true, false) => BatchChangeStatus.Failed
+          case (false, true, true) => BatchChangeStatus.PartialFailure
+          case (false, true, false) => BatchChangeStatus.Failed
```
Comment (pauljamescleary) [inline modules/core/src/main/scala/vinyldns/core/domain/batch/BatchChange.scala]: I explicitly called out `false` here because parsing these 3 booleans in the pattern match was hard to follow.  The first pattern that says `true` makes the next two a moot point, but that was _sooo_ easy to miss I had to stare at this longer than I needed to.

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala (hunk 1 above)
Comment (britneywright) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: I’d go with the ignore allowManualReview

Comment (rebstar6) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: i disagree, give them the errors if there are any.

the point of being able to set allowManualReview=false is so that people can see/addess their errors. if that is ignored, they dont have that power

Comment (pauljamescleary) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: @rebstar6 can add this as an upfront validation

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 3]
```diff
@@ -355,47 +359,73 @@ class BatchChangeService(
...
+    val allErrors = transformed
+      .collect {
+        case Invalid(e) => e
+      }
+      .flatMap(_.toList)
+
+    // Tells us that we have soft errors, and no errors are hard errors
+    val hardErrorsPresent = allErrors.exists(_.isFatal)
+    val noErrors = allErrors.isEmpty
+    val isScheduled = batchChangeInput.scheduledTime.isDefined && this.scheduledChangesEnabled
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: `this.` isn't needed

Comment (pauljamescleary) [reply modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: @mitruly it is a little of setting new precedent, but it is hard with a file this humongous to know where all these variables are coming from.  I liked using `this` to identify it as a class level variable.
